### PR TITLE
Gh 820 export import buttons tests

### PR DESCRIPTION
**Pull request by macenturalxl1**
Hi here are tests to go along side the import & export buttons.

For import(), I couldn't test cover the lines for reader.onloadend function() - do you have any recommendations for that or is acceptable to not cover it?

Also I made the mistake of pulling the latest changes from develop branch instead of the gh-820 branch - if you pull the latest develop branch commits in to gh-820 then you should see it's just the one test file I've created, or if you prefer I can got back and revert the latest develop changes merge in my branch to be in sync with gh-820 commits.

Or I can direct the PR to develop and put a note on to wait for gh-820 merge.

Let me know what you would like me to do? Also any feedback, thanks

### CODE REVIEW

File: ui/src/main/webapp/app/toolbar/import-export-dialog/import-export-dialog.html [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+
+        Import your results from a Gaffer JSON file.
+        <input type="file" id="import-results-file" name="file" ng-model="importFilename"/>
+        <div style="display:flex; justify-content:flex-end; width:100%; padding:0;">
+            <md-button class="md-raised" ng-click="import()">Import</md-button>
+        </div>
+
+        Export your results to a Gaffer JSON file.
+        <div style="display:flex; justify-content:flex-end; width:100%; padding:0;">
```
Comment (d47853) [inline ui/src/main/webapp/app/toolbar/import-export-dialog/import-export-dialog.html:13]: Could you move this styling into a css class

**Comment by t11947** [comment]
> For import(), I couldn't test cover the lines for reader.onloadend function() - do you have any recommendations for that or is acceptable to not cover it?

Ideally I'd like this to be covered. Feel free to refactor the code to make it easier to test.

**Comment by t11947** [comment]
> Also I made the mistake of pulling the latest changes from develop branch instead of the [gh-820](https://github.com/gchq/gaffer-tools/issues/820) branch - if you pull the latest develop branch commits in to [gh-820](https://github.com/gchq/gaffer-tools/issues/820) then you should see it's just the one test file I've created, or if you prefer I can got back and revert the latest develop changes merge in my branch to be in sync with [gh-820](https://github.com/gchq/gaffer-tools/issues/820) commits.
> 
> Or I can direct the PR to develop and put a note on to wait for [gh-820](https://github.com/gchq/gaffer-tools/issues/820) merge.

We've resolved this. You're now merging into develop.

File: ui/src/main/webapp/app/toolbar/import-export-dialog/import-export-dialog.html (hunk 1 above)
Comment (macenturalxl1) [reply ui/src/main/webapp/app/toolbar/import-export-dialog/import-export-dialog.html]: Done - I've reused the CSS and tagging that is already available - also enhanced the design of the modal slightly that is more in line with the rest of the app design (in terms of formatting and colouring)

**Comment by macenturalxl1** [comment]
> We've resolved this. You're now merging into develop.

Thanks

**Comment by macenturalxl1** [comment]
> > For import(), I couldn't test cover the lines for reader.onloadend function() - do you have any recommendations for that or is acceptable to not cover it?
> 
> Ideally I'd like this to be covered. Feel free to refactor the code to make it easier to test.

Agreed - I'll refactor the code so the block is testable...

**Comment by macenturalxl1** [comment]
Hi - ready for next review.

CSS comment on the html view - I've replaced the inline styling and enhanced the design using existing CSS and tags available so it is more consitent with the overall application design. It also means the styling is now dynamic, if multiple themes are added in future (i.e. dark mode, other custom CSS). Let me know what you think of the improved design - it is in a state that is easy to change to add your own ideas.

Test coverage - I tried multiple ways to cover the block, however found the cleanest and safest way to both test it and allow the FileReader onload to access it was to add it to the $scope. I tried adding it as a component or binding it to `this` etc. but required more complex and unreadable implementation. Adding to scope just seemed to be the better option.

Let me know what you think?

File: ui/src/main/webapp/app/toolbar/import-export-dialog/import-export-dialog.html (hunk 1 above)
Comment (macenturalxl1) [reply ui/src/main/webapp/app/toolbar/import-export-dialog/import-export-dialog.html]: Hi @d47853 - is this CSS change ok?

**Pull request merged**